neo4j/check.py

```diff
@@ -0,0 +1,148 @@
+import time
+import requests
+
+# stdlib
+import socket
+
+# 3rd party
+from httplib2 import HttpLib2Error
+import simplejson as json
+import requests
+from hashlib import md5
+import base64
+
+# project
+from checks import AgentCheck
+from util import headers
+
+
+class HTTPCheck(AgentCheck):
+    SERVICE_CHECK_NAME = 'neo4j.can_connect'
+	
+	# Neo4j metrics to send
+    keys = set([
+        'kernelversion',
+        'storeid',
+        'storecreationdate',
+        'storelogversion',
+        'kernelstarttime',
+        'lastcommittedtxid',
+        'peaknumberofconcurrenttransactions',
+        'numberofrolledbacktransactions',
+        'numberofopentransactions',
+        'numberofopenedtransactions',
+        'numberofcommittedtransactions',
+        'logicallogsize',
+        'propertystoresize',
+        'arraystoresize',
+        'totalstoresize',
+        'relationshipstoresize',
+        'stringstoresize',
+        'nodestoresize',
+        'locks',
+        'numberofaverteddeadlocks',
+        'numberofrelationshipidsinuse',
+        'numberofpropertyidsinuse',
+        'numberofnodeidsinuse',
+        'numberofrelationshiptypeidsinuse',
+        'memorypools',
+        'pins',
+        'evictions',
+        'byteswritten',
+        'filemappings',
+        'fileunmappings',
+        'bytesread',
+        'flushes',
+        'evictionexceptions',
+        'faults',
+        'ha.pull_interval',
+        'dbms.memory.pagecache.size',
+    ])
+	
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        AgentCheck.__init__(self, name, init_config, agentConfig, instances)
+
+	
+	
+    def check(self, instance):
+        host, port, user, password, connect_timeout, server_name,version = self._get_config(instance)
+        usrPass = user + ":" + password
+        b64Val = base64.b64encode(usrPass)
+        url = instance['neo4j_url'] + ":" + str(port) + "/"
+        if int(version[0]) > 2:
+        	checkURL = host + ":" + str(port) + "/db/data/transaction/commit"
+        else:
+	        checkURL = host + ":" + str(port) + "/v1/service/metrics"
+        print checkURL
+        tags = instance.get('tags', [])
+        service_check_tags = tags + ['url:%s' % url]
+      
+
+# Neo specific
+# Create payload using built-in Neo4j queryJmx stored procedure
+        try:
+            payload = {"statements" : [{"statement" : "CALL dbms.queryJmx('org.neo4j:*') yield attributes with  keys(attributes) as k, attributes unwind k as row return row, attributes[row]['value'];"}]}
+            headers_sent = {'Content-Type':'application/json','Authorization':'Basic ' + b64Val + '','Content-Type':'application/json'}
+            r = requests.post(checkURL, data=json.dumps(payload),headers=headers_sent)
+
+        except (socket.timeout, socket.error, HttpLib2Error) as e:
```

Comment:
Is this required? The check relies on request which does not make use of HttpLib2, as such I don't believe we'd ever hit this exception.  Dropping this would also allow us to drop httplib2 as a dependency.

Reply:
I think we can remove httplib2.